primitives/src/runtime_api.rs

```diff
@@ -123,7 +123,7 @@ use sp_std::{collections::btree_map::BTreeMap, prelude::*};
 
 sp_api::decl_runtime_apis! {
 	/// The API for querying the state of parachains on-chain.
-	#[api_version(4)]
+	#[api_version(5)]
```

Comment:
This is good. You also need to add (can't leave a comment at the right):
```
#[changed_in(5)]
fn availability_cores() -> Vec<CoreState<H, N>>;
```
This way you can have two implementations - old and new ones. The intention of `changed_in` is to change the method signature. I'm not sure if it will work in the case 'same signature, different implementation'.

If this is the case - you can just add another function and handle the right one in the client code. This one breaks the semantics a little but imo it's fine. If pick this approach, you shouldn't touch base version here and just add another method:
```
#[api_version(5)]
fn availability_cores_with_parathreads() -> Vec<CoreState<H, N>>;
```